api/envoy/config/route/v3/route_components.proto

```diff
@@ -736,7 +765,38 @@ message RouteAction {
   //
   //   Having above entries in the config, requests to */prefix* will be stripped to */*, while
   //   requests to */prefix/etc* will be stripped to */etc*.
-  string prefix_rewrite = 5;
+  string prefix_rewrite = 5 [(udpa.annotations.field_migrate).oneof_promotion = "path_rewrite"];
+
+  // Indicates that during forwarding, portions of the path that match the
+  // pattern should be rewritten, even allowing the substitution of capture
+  // groups from the pattern into the new path as specified by the rewrite
+  // substitution string. This is useful to allow application paths to be
+  // rewritten in a way that is aware of segments with variable content like
+  // identifiers. The router filter will place the original path as it was
+  // before the rewrite into the :ref:`x-envoy-original-path
+  // <config_http_filters_router_x-envoy-original-path>` header.
+  //
+  // Only one of :ref:`prefix_rewrite <envoy_api_field_config.route.v3.RouteAction.prefix_rewrite>`
+  // or *regex_rewrite* may be specified.
+  //
+  // Examples:
+  //
+  // * The path pattern ``^/service/([^/]+)(/.*)$`` paired with a substitution
+  //   string of ``\2/instance/\1`` would transform ``/service/foo/v1/api``
+  //   into ``/v1/api/instance/foo``.
+  //
+  // * The pattern ``one`` paired with a substitution string of ``two`` would
+  //   transform ``/xxx/one/yyy/one/zzz`` into ``/xxx/two/yyy/two/zzz``.
+  //
+  // * The pattern ``^(.*?)one(.*)$`` paired with a substitution string of
+  //   ``\1two\2`` would replace only the first occurrence of ``one``,
+  //   transforming path ``/xxx/one/yyy/one/zzz`` into ``/xxx/two/yyy/one/zzz``.
+  //
+  // * The pattern ``(?i)/xxx/`` paired with a substitution string of ``/yyy/``
+  //   would do a case-insensitive match and transform path ``/aaa/XxX/bbb`` to
+  //   ``/aaa/yyy/bbb``.
+  RegexRewrite regex_rewrite = 32
+      [(udpa.annotations.field_migrate).oneof_promotion = "path_rewrite"];
```

Comment:
Sure.  I notice there's already a `path_rewrite_specifier` in the `RedirectAction` section.

Reply:
I think that's fine for that to be repeated in different types.

It does strike me that someone will eventually want to use RegexRewrite in RedirectAction.

@mattklein123: should we move RegexWrite so that it's not nested within RouteAction? And if so, the name might need to be a little more specific to paths, like RegexPathRewrite.

(@jphx sorry again for this piecemeal approach to the API review -- the maintainer with the best vision for this is currently unavailable.)

Reply:
Not a problem.  I'm willing to make whatever changes are best for Envoy (within reason!).

Reply:
We seem to have stalled on this review.  Is there anything I can do to help?  Are we waiting for the maintainer with the best vision to be available again?

If desired, I can move the `RegexRewrite` message out of the `RouteAction` message so that it can be more easily re-used in the `RedirectAction` message.  Or it could be moved out of the file altogether.

Reply:
I don't think we are stalled. @lizan had a comment below. Can you take a look at that and we can move forward?

Reply:
His comment was asking if the `RegexRewrite` message should be moved out of the file altogether and made a generic type.  It was my impression he was looking for the opinion of other maintainers, since it is a strategy question.  Surely you don't want _me_ to decide the answer to that question.

If you all agree this is what should be done, then I would be happy to make that change.